src/systems/a320_systems/src/hydraulic.rs

```diff
@@ -537,9 +566,25 @@ struct A320EngineDrivenPumpController {
 }
 impl A320EngineDrivenPumpController {
     const MIN_ENGINE_OIL_PRESS_THRESHOLD_TO_INHIBIT_FAULT: f64 = 18.;
+    const YELLOW_EDP_POWER_BUS1: ElectricalBusType = ElectricalBusType::DirectCurrent(2);
+    const YELLOW_EDP_POWER_BUS2: ElectricalBusType = ElectricalBusType::DirectCurrentEssential;
+    const GREEN_EDP_POWER_BUS: ElectricalBusType = ElectricalBusType::DirectCurrentEssential;
 
     fn new(engine_number: usize) -> Self {
+        let power_buses;
+        if engine_number == 1 {
+            power_buses = vec![Self::GREEN_EDP_POWER_BUS];
+        } else if engine_number == 2 {
+            power_buses = vec![Self::YELLOW_EDP_POWER_BUS1, Self::YELLOW_EDP_POWER_BUS2];
+        } else {
+            panic!("The A320 only supports two engines.");
+        }
+
+        assert!(!power_buses.is_empty() && power_buses.len() <= 2);
```

Comment:
**suggestion**: Pass these in via the constructor to save yourself this `if else if else` hassle.
```rust
A320EngineDrivenPumpController::new(1, vec![ElectricalBusType::DirectCurrentEssential]);
A320EngineDrivenPumpController::new(2, vec![ElectricalBusType::DirectCurrent(2), ElectricalBusType::DirectCurrentEssential]);
```